src/AI/Muriel.java

```diff
@@ -331,18 +330,18 @@ public GameAction getNextAction(GameMap gameMap)
         for( GameAction action : attackActions )
         {
           // Sift through all attack actions we can perform.
-          XYCoord targetLoc = action.getTargetLocation();
-          Unit target = gameMap.getLocation(targetLoc).getResident();
-          Environment environment = gameMap.getEnvironment(targetLoc);
+          double damageValue = AIUtils.scoreAttackAction(unit, action, gameMap,
+              (results) -> {
+                double hpDamage = Math.min(results.defenderHPLoss, results.defender.getPreciseHP());
 
-          // Calculate the cost of the damage we can do.
-          double attackDamage = CombatEngine.calculateOneStrikeDamage(unit, 1, target, gameMap, environment.terrainType.getDefLevel(), unit.model.hasMobileWeapon());
+                if( shouldAttack(unit, results.defender, gameMap) )
+                  return (results.defender.model.getCost() / 10) * hpDamage;
 
-          double hpDamage = Math.min(attackDamage, target.getPreciseHP());
-          double damageValue = (target.model.getCost()/10) * hpDamage;
+                return 0.;
+              }, (terrain, params) -> 0.01); // Attack terrain, but don't prioritize it over units
```

Comment:
This gives attacking terrain too much importance. You think I'm kidding, but it totally derails units who have more important things to do and shouldn't be messing with meteors in the first place. I think I'd rather Muriel just deny the existence of meteors than do this.

Reply:
Okay.